src/main/kotlin/com/exactpro/th2/crawler/state/BaseState.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.crawler.state
+
+import com.exactpro.th2.dataprovider.grpc.DataProviderService
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+
+
+/**
+ * This interface is used to mark the recovery state class and choose the right one during deserialization
+ */
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    property = "version",
+    include = JsonTypeInfo.As.PROPERTY,
+)
+interface BaseState
+
+/**
+ * The version of the recovery state that can be used to determinate the conversions that should be applied to get the current state version
+ */
+interface VersionMarker {
+    val name: String
+    val number: Int
+}
+
+/**
+ * The state versions. When new version is added it should be last one in the enum list.
+ *
+ * The order is important. It will be used to determinate which conversions should be made
+ */
+enum class Version : VersionMarker {
+    V_1;
+
+    override val number: Int = ordinal
+}
+
+interface StateProvider {
+    val version: VersionMarker
```

Comment:
Maybe add the target version additionally?

Reply:
I think it is redundant because the target version is the next after the `version`. If we need to change the order we can update the enum order or `number` field in the `VersionMarker`

Reply:
VersionMarker is an interface and we can implement any sequence of versions, for example 1, 5, 10